internal/provider/provider.go

```diff
@@ -80,23 +85,58 @@ func (p *HumanitecProvider) Metadata(ctx context.Context, req provider.MetadataR
 func (p *HumanitecProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
+			"hctl_config_file": schema.StringAttribute{
+				MarkdownDescription: "Path to the hctl config file path. Takes precedences over the HUMANITEC_ environment variables.",
+				Optional:            true,
+			},
 			"api_url": schema.StringAttribute{
-				MarkdownDescription: "Humanitec API URL prefix",
+				MarkdownDescription: "Humanitec API URL prefix. Takes precedence over the contents of hctl_config_file but overridden by the HUMANITEC_API_PREFIX environment variable.",
 				Optional:            true,
 			},
 			"org_id": schema.StringAttribute{
-				MarkdownDescription: "Humanitec Organization ID",
+				MarkdownDescription: "Humanitec Organization ID. Takes precedence over the contents of hctl_config_file but overridden by the HUMANITEC_ORG environment variable.",
 				Optional:            true,
 			},
 			"auth_token": schema.StringAttribute{
-				MarkdownDescription: "Humanitec Auth Token",
+				MarkdownDescription: "Humanitec Auth Token. Takes precedence over the contents of hctl_config_file but overridden by the HUMANITEC_AUTH_TOKEN environment variable.",
 				Sensitive:           true,
 				Optional:            true,
 			},
 		},
 	}
 }
 
+type Config struct {
+	HctlConfigFile string `yaml:"hctl_config_file" json:"hctl_config_file"`
+	ApiUrl         string `yaml:"api_url" json:"api_url"`
+	DefaultOrg     string `yaml:"default_org_id" json:"default_org_id"`
+	Token          string `yaml:"token" json:"token"`
+}
+
+func readConfigFile(path string) (Config, error) {
+	var cfg Config
+	f, err := os.ReadFile(filepath.Clean(path))
+	if err != nil {
+		if os.IsNotExist(err) {
+			// If the file does not exist, return an empty config
+			return cfg, nil
+		}
+		return cfg, fmt.Errorf("failed to read config file: %w", err)
+	}
+	if err := yaml.Unmarshal(f, &cfg); err != nil {
+		return cfg, fmt.Errorf("failed to parse config file: %w", err)
+	}
+	return cfg, nil
+}
+
+func getConfigFilePath() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", fmt.Errorf("failed to get user home directory: %w", err)
+	}
+	return path.Join(homeDir, ".config", "hctl", "config.yaml"), nil
+}
```

Comment:
Suggesting to use ready config method `os.UserConfigDir()`:
```suggestion
func getConfigFilePath() (string, error) {
	configDir, err := os.UserConfigDir()
	if err != nil {
		return "", fmt.Errorf("failed to get user config home directory: %w", err)
	}
	return filepath.Join(configDir, "hctl", "config.yaml"), nil
}
```
plus switch to `filepath.Join()` vs `path.Join()`